lib/switch-header-source.coffee

```diff
@@ -2,6 +2,22 @@ fs   = require 'fs-plus'
 path = require 'path'
 
 module.exports =
+  config:
+    headerFileRegex:
+      type: 'string'
+      default: '\\.h|\\.hpp|\\.hh|\\.hxx'
+      description: """Regular expression used to identify "header" file
+                      suffixes (matched at the end of the file name, remember
+                      to escape \'.\')"""
+      order: 1
+    definitionFileRegex:
+      type: 'string'
+      default: '\\.c|\\.cpp|\\.cc|\\.cxx|\\.m|\\.mm'
```

Comment:
Profiler shows that this regexp needs 20x more time to evaluate than the header regexp for a *.C <-> *.h pair. I think this is due to the case insensitive flag on the regexp. That should not be there. Case matters in extensions (*.c vs. *.C for example).